activerecord/lib/active_record/migration.rb

```diff
@@ -1092,10 +1088,45 @@ def pending_migrations
     end
 
     def migrated
-      @migrated_versions ||= Set.new(self.class.get_all_versions)
+      @migrated_versions || load_migrated
```

Comment:
It's explicitly set to nil in Migrator#initialize. Do we need a `defined?` in this case?